### PR TITLE
Fix double spinners

### PR DESCRIPTION
**Pull request by alexweininger**
Depends on https://github.com/microsoft/vscode-azuretools/pull/1156
Alternative to #230 

Here's how it looks:

https://user-images.githubusercontent.com/12476526/167943950-06bb1765-9062-45ac-beac-43d2b18fe972.mov

### CODE REVIEW

File: src/tree/ResolvableTreeItemBase.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
     public abstract parent?: AzExtParentTreeItem | undefined;
     public abstract fullId: string;
 
+    // Setting this forces the tree item to always start out with a spinner icon, and have a "Loading..." description
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    private _temporaryDescription: string | undefined = 'Loading...';
```
Comment (alexweininger) [inline src/tree/ResolvableTreeItemBase.ts:28]: I know this is a hack, but if we don't make the loading state the default, then it results in some flashing.

Comment (nturinski) [inline src/tree/ResolvableTreeItemBase.ts:28]: Why is this necessary?  Because of the refresh?

File: src/tree/ResolvableTreeItemBase.ts [hunk 2]
```diff
@@ -53,7 +58,10 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
 
             const resolver = this.getResolver();
 
-            await this.runWithTemporaryDescription(context, 'Loading...', async () => {
+            await this.runWithTemporaryDescription(context, {
+                description: 'Loading...',
```
Comment (nturinski) [inline src/tree/ResolvableTreeItemBase.ts:62]: Should probably go ahead and just localize this.

File: src/tree/ResolvableTreeItemBase.ts (hunk 1 above)
Comment (alexweininger) [reply src/tree/ResolvableTreeItemBase.ts:28]: See above comment

File: src/tree/ResolvableTreeItemBase.ts [hunk 3]
```diff
@@ -22,6 +25,11 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
     public abstract parent?: AzExtParentTreeItem | undefined;
     public abstract fullId: string;
 
+    // Setting this forces the tree item to always start out with a spinner icon, and have a "Loading..." description
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    private _temporaryDescription: string | undefined = loading;
```
Comment (bwateratmsft) [inline src/tree/ResolvableTreeItemBase.ts:31]: image

File: src/tree/ResolvableTreeItemBase.ts [hunk 4]
```diff
@@ -53,7 +61,10 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
 
             const resolver = this.getResolver();
 
-            await this.runWithTemporaryDescription(context, 'Loading...', async () => {
+            await this.runWithTemporaryDescription(context, {
+                description: loading,
+                skipRefresh: true
```
Comment (bwateratmsft) [inline src/tree/ResolvableTreeItemBase.ts:66]:
```suggestion
                softRefresh: true
```

In the utils package it's `softRefresh` (which I think is a better name as it's still doing _some_ refresh)

File: src/tree/ResolvableTreeItemBase.ts [hunk 5]
```diff
@@ -53,7 +61,10 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
...
+            }, async () => {
                 this.resolveResult = await resolver.resolveResource(this.subscription, this.data);
 
                 // Debug only?
```
Comment (bwateratmsft) [inline src/tree/ResolvableTreeItemBase.ts:70]: There's a comment below (but GitHub won't let me add it in context):
> `// It is not needed to refresh at this point, because runWithTemporaryDescription already does that`

Can we make sure that resolving can update the label/description? I _think_ so but I'm not totally sure. Stopping a function app makes resolving update the description so IMO that's a sufficient test--if that "Stopped" description still appears I'm happy.

**Review by bwateratmsft — APPROVED** [review]

File: src/tree/ResolvableTreeItemBase.ts (hunk 5 above)
Comment (bwateratmsft) [reply src/tree/ResolvableTreeItemBase.ts:70]: :chef's kiss:
Perfect! Also, love how fast SWA resolves.

File: src/tree/ResolvableTreeItemBase.ts (hunk 3 above)
Comment (alexweininger) [reply src/tree/ResolvableTreeItemBase.ts:31]: Was there an action Item here or are we just celebrating Frozen 2?

Comment (bwateratmsft) [reply src/tree/ResolvableTreeItemBase.ts:31]: The second one (just celebrating Frozen 2)

Comment (alexweininger) [reply src/tree/ResolvableTreeItemBase.ts:31]: okbye
